src/ArrowFunctions.cpp

```diff
@@ -275,25 +275,47 @@ int64_t cpp_getStringColumnNumBytes(const char* filename, const char* colname, v
         }
         column_reader = row_group_reader->Column(idx);
 
-        int16_t definition_level;
         parquet::ByteArrayReader* ba_reader =
           static_cast<parquet::ByteArrayReader*>(column_reader.get());
 
         int64_t numRead = 0;
         while (ba_reader->HasNext() && numRead < numElems) {
-          parquet::ByteArray value;
-          (void)ba_reader->ReadBatch(1, &definition_level, nullptr, &value, &values_read);
-          if ((ty == ARROWLIST && definition_level == 3) || ty == ARROWSTRING) {
-            if(values_read > 0) {
-              offsets[i] = value.len + 1;
-              byteSize += value.len + 1;
-              numRead += values_read;
-            } else {
-              offsets[i] = 1;
-              byteSize+=1;
-              numRead+=1;
+          if((numElems - numRead) < batchSize)
+            batchSize = numElems-numRead;
+          std::vector<parquet::ByteArray> string_values(batchSize);
+          std::vector<int16_t> definition_level(batchSize);
+          (void)ba_reader->ReadBatch(batchSize, definition_level.data(), nullptr, string_values.data(), &values_read);
+          numRead += values_read;
+          if (ty == ARROWSTRING) {
+            auto numCols = file_metadata -> num_columns();
+            int string_index = 0;
+            for(int idx = 0; idx < definition_level.size(); idx++) {
+              auto lvl = definition_level[idx];
+              auto value = string_values[string_index];
+              if(lvl != 0 || numCols > 1) {
+                auto value = string_values[string_index];
```

Comment:
You've caught me redhanded... this was a fragment of something that I was trying earlier, where I was basing one of the conditionals on `value`, but couldn't do that since `value` could potentially be uninitialized.

Will remove that first value set.